### PR TITLE
[Dev] Support loading Emission from AppHub

### PR DESCRIPTION
**Pull request by orta**
Adds a new section when in the deploy schema (e.g. you will not see this in dev)

screen shot 2017-05-17 at 11 10 40

It has the same build chooser that Eigen has:

screen shot 2017-05-17 at 11 10 45

And can pull out the related PR from the build:

screen shot 2017-05-17 at 11 10 50

---

This makes the apphub upload also include the PR metadata as a description, so there should be more builds that look like:

screen shot 2017-05-17 at 11 12 31

--- 

I disabled the tests part of our Pods setup. Until we're writing native tests it adds more complications to the build process for everyone - and I was having some tricky issues with the React subspecs and AppHub.<hr data-danger="yep"/>

### Tested on Device?

- [ ] @orta
- [ ] @alloy

<details>
  <summary>How to get set up with this PR?</summary>
  <p>&nbsp;</p>
   <p><b>To run on your computer:</b></p>
<pre><code>git fetch origin pull/541/head:orta-541-checkout
git checkout orta-541-checkout
yarn install
cd example; pod install; cd ..
open -a Simulator
yarn start</code></pre>
   </p>
   <p>Then run <code>xcrun simctl launch booted net.artsy.Emission</code> once a the simulator has finished booting</p>
   <p><b>To run inside Eigen (prod or beta) or Emission (beta):</b> Shake the phone to get the Admin menu.</p>
   <p>If you see <i>"Use Staging React Env" </i> - click that and restart, then follow the next step.</p>
   <p>Click on <i>"Choose an RN build" </i> - then pick the one that says: "X,Y,Z"</p>
   <p>Note: this is a TODO for PRs, currently  you can only do it on master commits.</p>
</details>

### CODE REVIEW

**Comment by orta** [comment]
Current build now shows relative time, I don't really know what day it is, so I shouldn't expect others to.
screen shot 2017-05-17 at 11 59 59

**Comment by alloy** [comment]
Going to look at this after our current release woes are over.

**Comment by orta** [comment]
Yah, that's fine, less changing it 👍

**Comment by orta** [comment]
poke

File: Example/Emission/AppDelegate.m [hunk 1]
```diff
@@ -103,7 +103,7 @@ - (void)setupEmissionWithUserID:(NSString *)userID accessToken:(NSString *)acces
                           useStagingEnvironment:useStaging
                                       sentryDSN:nil];
 #else
-#ifdef DEPLOY
+#if DEPLOY
```
Comment (alloy) [inline Example/Emission/AppDelegate.m]: Why do you use `ifdef` in the file above but prefer `if` here?

File: Example/Emission/NSDateFormatter+TimeAgo.m [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    if (diff == 1) return @"1 hour ago";
+
+    return [NSString stringWithFormat:@"%@ hours ago", @(diff)];
+
+  } else if (time < 604800) {
+    NSInteger diff = round(time / 60 / 60 / 24);
+    if (diff == 1) return @"yesterday";
+    if (diff == 7) return @"last week";
+
+    return[NSString stringWithFormat:@"%@ days ago", @(diff)];
```
Comment (alloy) [inline Example/Emission/NSDateFormatter+TimeAgo.m]: Nit: missing whitespace

File: Example/Podfile [hunk 3]
```diff
@@ -24,7 +24,7 @@ target 'Emission' do
   pod 'Artsy+UIFonts'
 
   # For easy updating of the JS
-  pod 'AppHub'
+  pod 'AppHub', :git => 'https://github.com/orta/apphub.git', :branch => "build_list"
```
Comment (alloy) [inline Example/Podfile]: Good fix 👍

File: Example/Podfile [hunk 4]
```diff
@@ -40,14 +40,14 @@ target 'Emission' do
...
-    # pod 'Expecta'
-  end
+# target 'EmissionTests' do
+#    inherit! :search_paths
+#
+#    pod 'React', :path => react_path, :subspecs => %w(Core RCTTest)
+#
+#    pod 'Specta'
+#    pod 'Expecta'
+# end
```
Comment (alloy) [inline Example/Podfile]: Fair 👍

File: scripts/deploy_apphub.sh [hunk 5]
```diff
@@ -8,9 +8,16 @@ npm install --global apphubdeploy
 # Create a .apphub credentials file
 echo "{\"appHubId\":\"Z6IwqK52JBXrKLI4kpvJ\",\"appHubSecret\":\"$APP_HUB_SECRET\"}" > .apphub
 
+# Get the most recent PR commit
+SHA=`git rev-list --min-parents=2 --max-count=1 HEAD`
+# Pull the name of the PR out of the auto-generated commit description
+PR_DESC=`git log --format=%B -n 1 $SHA | tail -1`
+# This'll break when we hit 1k PRs, but that's still a while away and an easy fix
+PR_NUM=`git log --format=%B -n 1 $SHA | grep -o '#[0-9][0-9][0-9]' | tail -n 1`
```
Comment (alloy) [inline scripts/deploy_apphub.sh]: Is there a reason you can’t use `grep -Eo '#[0-9]+'` instead?

Comment (orta) [inline scripts/deploy_apphub.sh]: If that works - even better :D

File: Example/Emission/AppDelegate.m (hunk 1 above)
Comment (orta) [inline Example/Emission/AppDelegate.m]: I might switch the app entirely to just debug and release, I don't think a third schema is useful as release will not ever be used

Comment (alloy) [inline Example/Emission/AppDelegate.m]: I use release sometimes to test on a device.

File: scripts/deploy_apphub.sh (hunk 5 above)
Comment (alloy) [inline scripts/deploy_apphub.sh]: Yup, the `-E` is for BSD grep to support extended regexp, I believe that even GNU grep secretly supports the `-E` option to be compatible with BSD grep.

File: Example/Emission/AppDelegate.m (hunk 1 above)
Comment (orta) [inline Example/Emission/AppDelegate.m]: Alright - I'll keep it around 👍

**Comment by orta** [comment]
Updated and addressed all feedback

**Comment by alloy** [comment]
Ace 🙏🏽

**Pull request merged**